Programs don't build for GOOS=linux GOARCH=arm GOARM=5

**Current behavior (how does the issue manifest):**
Get error when trying to build program for the indicated platform:

GOOS=linux GOARCH=arm GOARM=5 flogo build -e
# ev3rules/vendor/github.com/project-flogo/rules/rete
vendor/github.com/project-flogo/rules/rete/network.go:342:14: constant 4294967295 overflows int
FATAL: command "build" failed: exit status 2

**Expected behavior:**
Build should complete.

**Minimal steps to reproduce the problem (not required if feature enhancement):**
Try building any program using Flogo rules for the above platform 

**Please tell us about your environment (Operating system, docker version, browser & web ui version, etc):**
GOOS=linux GOARCH=arm GOARM=5 

**Flogo Rules version (CLI & contrib/lib. If unknown, leave empty or state unknown):** 0.X.X

**Additional information you deem important (e.g. issue happens only occasionally):**

--- a/rete/network.go
+++ b/rete/network.go
@@ -353,7 +353,7 @@ func (nw *reteNetworkImpl) createJoinNodeFromExisting(rule model.Rule, nodesOfRu
 
 func (nw *reteNetworkImpl) createJoinNodeFromSome(rule model.Rule, nodesOfRule *list.List,
 	classNodeLinksOfRule *list.List, conditionSet *list.List, nodeSet *list.List) bool {
-	leastNeeded := math.MaxUint32
+	leastNeeded := math.MaxInt32
 	maxIdentifier := -1
 	var targetNode node
 	var targetCondition model.Condition

